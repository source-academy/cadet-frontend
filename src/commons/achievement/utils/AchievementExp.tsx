import { Icon } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import React from 'react';

type AchievementExpProps = {
  exp: number;
  isBonus: boolean;
};

function AchievementExp(props: AchievementExpProps) {
  const { exp, isBonus } = props;

  if (exp === 0) return <div className="exp"></div>;

  // Converts number to EXP string
  const stringifyExp = (exp: number) => {
    return (isBonus ? '+' : '') + exp + ' XP';
  };

  return (
    <div className="exp">
      <Icon icon={IconNames.BANK_ACCOUNT} />
      <p>{stringifyExp(exp)}</p>
    </div>
  );
}

export default AchievementExp;
