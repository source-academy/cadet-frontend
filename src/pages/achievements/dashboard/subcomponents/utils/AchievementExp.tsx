import { Icon } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import React from 'react';

type AchievementExpProps = {
  exp?: number;
};

function AchievementExp(props: AchievementExpProps) {
  const { exp } = props;

  // Converts number to EXP string
  const stringifyExp = (exp: number | undefined) => {
    return exp === undefined ? 'No XP' : exp + ' XP';
  };

  return (
    <div className="exp">
      <Icon icon={IconNames.BANK_ACCOUNT} />
      <p>{stringifyExp(exp)}</p>
    </div>
  );
}

export default AchievementExp;
