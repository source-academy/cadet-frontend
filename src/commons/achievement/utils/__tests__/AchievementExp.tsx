import { mount } from 'enzyme';
import React from 'react';

import AchievementExp from '../AchievementExp';

const expProps = {
  exp: 0
};

const undefinedExpProps = {
  exp: undefined
};

test('AchievementExp component renders correctly', () => {
  const expPresent = <AchievementExp {...expProps} />;
  const tree = mount(expPresent);
  expect(tree.debug()).toMatchSnapshot();

  const expAbsent = <AchievementExp {...undefinedExpProps} />;
  const tree_1 = mount(expAbsent);
  expect(tree_1.debug()).toMatchSnapshot();
});
