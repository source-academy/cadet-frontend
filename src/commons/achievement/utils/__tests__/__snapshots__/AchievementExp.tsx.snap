// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AchievementExp component renders correctly 1`] = `
"<AchievementExp exp={0} isBonus={false}>
  <div className=\\"exp\\" />
</AchievementExp>"
`;

exports[`AchievementExp component renders correctly 2`] = `
"<AchievementExp exp={[undefined]} isBonus={false}>
  <div className=\\"exp\\" />
</AchievementExp>"
`;
